router/pkg/metric/prom_metric_store.go

```diff
@@ -47,89 +45,170 @@ func NewPromMetricStore(logger *zap.Logger, meterProvider *metric.MeterProvider,
 }
 
 func (h *PromMetricStore) MeasureInFlight(ctx context.Context, attr ...attribute.KeyValue) func() {
-	var baseKeys []attribute.KeyValue
-
-	baseKeys = append(baseKeys, h.baseAttributes...)
-	baseKeys = append(baseKeys, attr...)
-
-	baseAttributes := otelmetric.WithAttributes(baseKeys...)
+	dimensionsSet, otherAttributes := filterStringSliceAttr(attr)
+	otherAttributes = append(otherAttributes, otherAttributes...)
+	attributeAddOpt := otelmetric.WithAttributes(otherAttributes...)
 
 	if c, ok := h.measurements.upDownCounters[InFlightRequestsUpDownCounter]; ok {
-		c.Add(ctx, 1, baseAttributes)
+		if len(dimensionsSet) == 0 {
+			c.Add(ctx, 1, attributeAddOpt)
+		} else {
+			// String Slice attributes have to be exploded into multiple metrics with different label values in Prometheus.
+			// This is because Prometheus does not support multi-value labels.
+			for _, attr := range dimensionsSet {
+				for _, v := range attr.Value.AsStringSlice() {
+					c.Add(ctx, 1, otelmetric.WithAttributes(append(otherAttributes, attribute.KeyValue{
+						Key:   attr.Key,
+						Value: attribute.StringValue(v),
+					})...))
+				}
+			}
```

Comment:
Can you pull this out into a function that we call with a value parameter? This double loop is repeated a few times throughout